fem/prestriction.cpp

```diff
@@ -27,68 +27,57 @@ ParL2FaceRestriction::ParL2FaceRestriction(const ParFiniteElementSpace &fes,
                                            ElementDofOrdering e_ordering,
                                            FaceType type,
                                            L2FaceValues m)
-   : fes(fes),
-     nf(fes.GetNFbyType(type)),
-     vdim(fes.GetVDim()),
-     byvdim(fes.GetOrdering() == Ordering::byVDIM),
-     ndofs(fes.GetNDofs()),
-     dof(nf>0 ?
-         fes.GetTraceElement(0, fes.GetMesh()->GetFaceBaseGeometry(0))->GetDof()
-         : 0),
-     m(m),
-     nfdofs(nf*dof),
-     scatter_indices1(nf*dof),
-     scatter_indices2(m==L2FaceValues::DoubleValued?nf*dof:0),
-     offsets(ndofs+1),
-     gather_indices((m==L2FaceValues::DoubleValued? 2 : 1)*nf*dof)
+   : L2FaceRestriction(fes, type, m)
 {
    if (nf==0) { return; }
    // If fespace == L2
-   const FiniteElement *fe = fes.GetFE(0);
+   const ParFiniteElementSpace &pfes =
```

Comment:
I don't have any problem with the changes in this file, but can you explain why they are part of this PR?

Reply:
Because I realized while working on this PR that this code could be simplified and avoid code duplication.

Reply:
Also, it is not mentioned in the description of the PR, but the `FactorizeBlocks` method has impact on Element Assembly, because "internal" face terms are added to elements matrices, and therefore don't require the extra step of applying them, therefore improving the performance of `Mult` for Element Assembly.

Reply:
I forgot to say that one of the main change is that `ParL2FaceRestriction` now inherit from `L2FaceRestriction`, preventing from duplicating the new methods `FillI`, `FillJAndData`, `AddFaceMatricesToElementMatrices`.